tests/integration/pure_cairo_vm/cheatcodes/expect_call/expect_call_test.cairo

```diff
@@ -0,0 +1,64 @@
+from starkware.cairo.common.math import assert_not_zero
+
+func test_expect_call_success(){
```

Comment:
Please add tests:
- [] Expect call through proxy
- [] Expect call in a remote contract (with the depth of call 2) -> expect call on contract C and test calls B which calls C
- [] expect call with selector instrad of str

Reply:
Adding as todos in the main PR comment